---
title: "PS2_template"
output:
  pdf_document: default
  html_document: default
date: "2025-04-25"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# RNA-seq Quality Assessment Assignment - Bi 623 (Summer 2025 Assignment/PS 2)

## Overall assignment:

In this assignment, you will process electric organ and/or skeletal muscle RNA-seq reads for a future differential gene expression analysis. We will be completing the differential gene expression analysis in our last bioinformatics assignment of this class. You will learn how to use existing tools for quality assessment and read trimming, compare quality assessments to those created by your own software, and how to align and count reads. Additionally, you will learn how to summarize important information in a high-level report. You should create a cohesive, well written report for your "PI" about what you've learned about/from your data.

**This template is provided as reference for instructions. Files with specific naming conventions are requested to be turned in at the end of this problem set. You can use this template to gather notes while completing this assignment.** Be sure to upload all relevant materials by the deadline and **double check** to be sure that your offline repository is up-to-date with your online repository. Answers to questions should be included in your final, high-level, report as a `pdf`. This pdf should be generated using Rmarkdown and submitted to Canvas as well as GitHub. Be sure to keep a well-organized, detailed lab notebook!


### Dataset:

Each of you will be working with 2 RNA-seq files from two different electric fish studies (PRJNA1005245 and PRJNA1005244). The methods for the PRJNA1005244 dataset are [published](https://doi.org/10.1093/molbev/msae021) and the methods for the PRJNA1005245 dataset are written in the third chapter of a [thesis](https://canvas.uoregon.edu/courses/266187/files/22059308?module_item_id=5380118). For all steps below, process the two libraries separately. SRR assignments are here: ```/projects/bgmp/shared/Bi623/PS2/QAA_data_Assignments.txt```. If you have time, consider claiming and processing additional RNA-seq raw sequencing files via this [google doc](https://docs.google.com/document/d/1vEmVEzUaTjbDF4JyNsWH-wFpi8dm4wkcvWgSoYZzoCY/edit?usp=sharing). Although this is not extra credit, it will make our downstream RNA-seq analysis more interesting and your classmates will appreciate your efforts.

You are responsible for downloading this data from NCBI SRA, dumping into FASTQ files, and zipping those files (check ICA1 for a refresher). We are processing this data for use in a future assignment, so please keep your files well organized. Finally, rename the files to the convention Species_sample_tissue_age/size_sample#_readnumber.fastq.gz.

**Reminder: This template file IS not your final product; however, it gives you a space to record all of the necessary information for your final report.**

```{bash, eval=FALSE}
## Download your data

mamba activate sra-toolkit_bgmp

prefetch SRR25630305

prefetch SRR25630397

fasterq-dump --split-files ./SRR25630305.sra

fasterq-dump --split-files ./SRR25630397.sra

gzip SRR25630305_1.fastq

gzip SRR25630305_2.fastq

gzip SRR25630397_1.fastq

gzip SRR25630397_2.fastq
```

## Part 1 – Read quality score distributions

1. Create a new conda environment called `QAA` and install `FastQC`, `cutadapt`, and `Trimmomatic`. Google around if you need a refresher on how to create conda environments. Recommend doing this in an interactive session, not the login node! Record details of how you created this environment in your lab notebook! Make sure you check your installation with:
   - `fastqc --version` (should be 0.12.1)  

[Record details on how you made the conda environment]

```{bash, eval=FALSE}

mamba create -n QAA

mamba activate QAA

mamba install fastqc

mamba install cutadapt

mamba install trimmomatic 

```

2. Using `FastQC` via the command line on Talapas, produce plots of the per-base quality score distributions for R1 and R2 reads. Also, produce plots of the per-base N content, and comment on whether or not they are consistent with the quality score plots.

[Include FastQC commands, plots of per-base N content, comments on consistency with quality score plots]

```{bash, eval=FALSE}

/usr/bin/time -v fastqc SRR25630305/*fastq.gz -o /projects/bgmp/ica/bioinfo/Bi623/PS2/QAA/SRR25630305

/usr/bin/time -v fastqc SRR25630397/*fastq.gz -o /projects/bgmp/ica/bioinfo/Bi623/PS2/QAA/SRR25630397


```
	Command being timed: "fastqc SRR25630305/SRR25630305_1.fastq.gz SRR25630305/SRR25630305_2.fastq.gz -o /projects/bgmp/ica/bioinfo/Bi623/PS2/QAA/SRR25630305"
	User time (seconds): 49.50
	System time (seconds): 2.49
	Percent of CPU this job got: 107%
	Elapsed (wall clock) time (h:mm:ss or m:ss): 0:48.30
	Average shared text size (kbytes): 0
	Average unshared data size (kbytes): 0
	Average stack size (kbytes): 0
	Average total size (kbytes): 0
	Maximum resident set size (kbytes): 496136
	Average resident set size (kbytes): 0
	Major (requiring I/O) page faults: 0
	Minor (reclaiming a frame) page faults: 179790
	Voluntary context switches: 15282
	Involuntary context switches: 75
	Swaps: 0
	File system inputs: 0
	File system outputs: 4624
	Socket messages sent: 0
	Socket messages received: 0
	Signals delivered: 0
	Page size (bytes): 4096
	Exit status: 0

	Command being timed: "fastqc SRR25630397/SRR25630397_1.fastq.gz SRR25630397/SRR25630397_2.fastq.gz -o /projects/bgmp/ica/bioinfo/Bi623/PS2/QAA/SRR25630397"
	User time (seconds): 65.35
	System time (seconds): 3.51
	Percent of CPU this job got: 108%
	Elapsed (wall clock) time (h:mm:ss or m:ss): 1:03.23
	Average shared text size (kbytes): 0
	Average unshared data size (kbytes): 0
	Average stack size (kbytes): 0
	Average total size (kbytes): 0
	Maximum resident set size (kbytes): 518916
	Average resident set size (kbytes): 0
	Major (requiring I/O) page faults: 0
	Minor (reclaiming a frame) page faults: 187449
	Voluntary context switches: 18220
	Involuntary context switches: 113
	Swaps: 0
	File system inputs: 0
	File system outputs: 4656
	Socket messages sent: 0
	Socket messages received: 0
	Signals delivered: 0
	Page size (bytes): 4096
	Exit status: 0

```{r, out.width='90%', fig.align='center'}

knitr::include_graphics("QAA_figs/Cco_com125_EO_6cm_1_per_base_n_content.png")
knitr::include_graphics("QAA_figs/Cco_com125_EO_6cm_1_per_base_quality.png")
knitr::include_graphics("QAA_figs/Cco_com125_EO_6cm_2_per_base_n_content.png")
knitr::include_graphics("QAA_figs/Cco_com125_EO_6cm_2_per_base_quality.png")
knitr::include_graphics("QAA_figs/Crh_rhy107_EO_adult_1_per_base_n_content.png")
knitr::include_graphics("QAA_figs/Crh_rhy107_EO_adult_1_per_base_quality.png")
knitr::include_graphics("QAA_figs/Crh_rhy107_EO_adult_2_per_base_n_content.png")
knitr::include_graphics("QAA_figs/Crh_rhy107_EO_adult_2_per_base_quality.png")

```


The near-zero N content is consistent with the high base-quality profiles. Bases are confidently called (not “N”) across the reads, including at the ends where quality dips slightly but remains high.

3. Run your quality score plotting script from your Demultiplexing assignment in Bi622. (Make sure you're using the "running sum" strategy!!) Describe how the `FastQC` quality score distribution plots compare to your own. If different, propose an explanation. Also, does the run time differ? Mem/CPU usage? If so, why?

```{r, out.width='90%', fig.align='center'}

knitr::include_graphics("QAA_figs/R1_Cco_com125_EO_6cm_mean_quality_scores.png")
knitr::include_graphics("QAA_figs/R2_Cco_com125_EO_6cm_mean_quality_scores.png")
knitr::include_graphics("QAA_figs/R1_Crh_rhy107_EO_adult_mean_quality_scores.png")
knitr::include_graphics("QAA_figs/R2_Crh_rhy107_EO_adult_mean_quality_scores.png")


```

The FastQC quality plots and my plots are similar. They all have very high scores at the beginning of that reads (~Q36) with a gradual decline toward the end of the reads, and R2 tails off slightly more than R1. Per-base N content is ~0% in FastQC, which matches the high quality seen in both sets of plots. Small visual differences are expected because FastQC shows median with IQR (box-and-whiskers) while mine plot the mean, FastQC bins later cycles (e.g., 105–109) which smooths the right side. Overall, the results are consistent; any offsets are methodological, not biological. FastQC ran a bit faster because it’s an optimized, compiled Java program, whereas my Python script is single-threaded and spends more time in gzip decompression and Python loops. FastQC typically uses more CPU and RAM but finishes sooner; the Python script uses less CPU/RAM but takes longer.

4. Comment on the overall data quality of your two libraries. Go beyond per-base qscore distributions. Examine the `FastQC` [documentation](https://www.bioinformatics.babraham.ac.uk/projects/fastqc/Help/3%20Analysis%20Modules/) for guidance on interpreting results and planning next steps. Make and justify a recommendation on whether these data are of high enough quality to use for further analysis. 

[Include comments on data quality and recommendation on whether this can be used for further analysis]

Both libraries meet the bar for downstream RNA-seq. Quality scores are high across most bases with only a small drop at the end. The N content is ~0% across positions, the Adapter Content panel doesn’t show a worrying rise, there aren’t big unknown “overrepresented” sequences, duplication levels are typical for RNA-seq (representing highly expressed transcripts), the GC content looks close to what we expect, and the read lengths are consistent. Overall, the data look clean and are good to use for downstream analysis.

## Part 2 – Adaptor trimming comparison

5.  If you haven't already in your QAA environment, install `Cutadapt` and `Trimmomatic`. Check your installations with:
    - `cutadapt --version` (should be 5.0)
    - `trimmomatic -version` (should be 0.39)

```{bash, eval=FALSE}

cutadapt --version = 5.1

trimmomatic -version = 0.40

```

6. Using `Cutadapt`, properly trim adapter sequences from your assigned files. Be sure to read how to use `Cutadapt`. Use default settings. What proportion of reads (both R1 and R2) were trimmed?

    <details>
    <summary>Try to determine what the adapters are on your own. If you cannot (or if you do, and want to confirm), click here to see the actual adapter sequences used.</summary>
  
    R1: `AGATCGGAAGAGCACACGTCTGAACTCCAGTCA`
    
    R2: `AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT`
    </details>

    - *Sanity check*: Use your Unix skills to search for the adapter sequences in your datasets and confirm the expected sequence orientations. Report the commands you used, the reasoning behind them, and how you confirmed the adapter sequences.
    
[Include commands and report out the proportion of reads trimmed]    

```{bash, eval=FALSE}

cutadapt -a AGATCGGAAGAGCACACGTCTGAACTCCAGTCA -A AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT -o Cco_com125_EO_6cm_1.trim.fastq.gz -p Cco_com125_EO_6cm_1.trim.fastq.gz Cco_com125_EO_6cm_1.fastq.gz Cco_com125_EO_6cm_2.fastq.gz

cutadapt -a AGATCGGAAGAGCACACGTCTGAACTCCAGTCA -A AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT -o Crh_rhy107_EO_adult_1.trim.fastq.gz -p Crh_rhy107_EO_adult_2.trim.fastq.gz Crh_rhy107_EO_adult_1.fastq.gz Crh_rhy107_EO_adult_2.fastq.gz


```

For Cco_com125_EO_6cm:
Read 1 - 18.2% reads trimmed
Read 2 - 18.5% reads trimmed

For Crh_rhy107_EO_adult:
Read 1 - 12.5% reads trimmed
Read 2 - 13.0% reads trimmed

	Command being timed: "cutadapt -a AGATCGGAAGAGCACACGTCTGAACTCCAGTCA -A AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT -o SRR25630305_1.trim.fastq.gz -p SRR25630305_2.trim.fastq.gz SRR25630305_1.fastq.gz SRR25630305_2.fastq.gz"
	User time (seconds): 44.04
	System time (seconds): 0.17
	Percent of CPU this job got: 94%
	Elapsed (wall clock) time (h:mm:ss or m:ss): 0:46.59
	Average shared text size (kbytes): 0
	Average unshared data size (kbytes): 0
	Average stack size (kbytes): 0
	Average total size (kbytes): 0
	Maximum resident set size (kbytes): 38132
	Average resident set size (kbytes): 0
	Major (requiring I/O) page faults: 0
	Minor (reclaiming a frame) page faults: 7525
	Voluntary context switches: 987
	Involuntary context switches: 35
	Swaps: 0
	File system inputs: 0
	File system outputs: 0
	Socket messages sent: 0
	Socket messages received: 0
	Signals delivered: 0
	Page size (bytes): 4096
	Exit status: 0
	
		Command being timed: "cutadapt -a AGATCGGAAGAGCACACGTCTGAACTCCAGTCA -A AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT -o SRR25630397_1.trim.fastq.gz -p SRR25630397_2.trim.fastq.gz SRR25630397_1.fastq.gz SRR25630397_2.fastq.gz"
	User time (seconds): 55.64
	System time (seconds): 0.81
	Percent of CPU this job got: 99%
	Elapsed (wall clock) time (h:mm:ss or m:ss): 0:56.75
	Average shared text size (kbytes): 0
	Average unshared data size (kbytes): 0
	Average stack size (kbytes): 0
	Average total size (kbytes): 0
	Maximum resident set size (kbytes): 38088
	Average resident set size (kbytes): 0
	Major (requiring I/O) page faults: 0
	Minor (reclaiming a frame) page faults: 7530
	Voluntary context switches: 431
	Involuntary context switches: 48
	Swaps: 0
	File system inputs: 0
	File system outputs: 0
	Socket messages sent: 0
	Socket messages received: 0
	Signals delivered: 0
	Page size (bytes): 4096
	Exit status: 0

7. Use `Trimmomatic` to quality trim your reads. Specify the following, **in this order**:
    - LEADING: quality of 3
    - TRAILING: quality of 3
    - SLIDING WINDOW: window size of 5 and required quality of 15
    - MINLENGTH: 35 bases

    Be sure to output compressed files and clear out all intermediate files.

```{bash, eval=FALSE}

/usr/bin/time -v trimmomatic PE \
-threads 4 \
/projects/bgmp/ica/bioinfo/Bi623/PS2/QAA/SRR25630305/SRR25630305_1.trim.fastq.gz \
/projects/bgmp/ica/bioinfo/Bi623/PS2/QAA/SRR25630305/SRR25630305_2.trim.fastq.gz \
/projects/bgmp/ica/bioinfo/Bi623/PS2/QAA/SRR25630305/SRR25630305_1.trim.paired.fastq.gz \
/projects/bgmp/ica/bioinfo/Bi623/PS2/QAA/SRR25630305/SRR25630305_1.trim.unpaired.fastq.gz \
/projects/bgmp/ica/bioinfo/Bi623/PS2/QAA/SRR25630305/SRR25630305_2.trim.paired.fastq.gz \
/projects/bgmp/ica/bioinfo/Bi623/PS2/QAA/SRR25630305/SRR25630305_2.trim.unpaired.fastq.gz \
HEADCROP:8 \
LEADING:3 \
TRAILING:3 \
SLIDINGWINDOW:5:15 \
MINLEN:35

/usr/bin/time -v trimmomatic PE \
-threads 4 \
/projects/bgmp/ica/bioinfo/Bi623/PS2/QAA/SRR25630397/SRR25630397_1.trim.fastq.gz \
/projects/bgmp/ica/bioinfo/Bi623/PS2/QAA/SRR25630397/SRR25630397_2.trim.fastq.gz \
/projects/bgmp/ica/bioinfo/Bi623/PS2/QAA/SRR25630397/SRR25630397_1.trim.paired.fastq.gz \
/projects/bgmp/ica/bioinfo/Bi623/PS2/QAA/SRR25630397/SRR25630397_1.trim.unpaired.fastq.gz \
/projects/bgmp/ica/bioinfo/Bi623/PS2/QAA/SRR25630397/SRR25630397_2.trim.paired.fastq.gz \
/projects/bgmp/ica/bioinfo/Bi623/PS2/QAA/SRR25630397/SRR25630397_2.trim.unpaired.fastq.gz \
HEADCROP:8 \
LEADING:3 \
TRAILING:3 \
SLIDINGWINDOW:5:15 \
MINLEN:35

```

TrimmomaticPE: Started with arguments:
 -threads 4 /projects/bgmp/ica/bioinfo/Bi623/PS2/QAA/SRR25630305/SRR25630305_1.trim.fastq.gz /projects/bgmp/ica/bioinfo/Bi623/PS2/QAA/SRR25630305/SRR25630305_2.trim.fastq.gz /projects/bgmp/ica/bioinfo/Bi623/PS2/QAA/SRR25630305/SRR25630305_1.trim.paired.fastq.gz /projects/bgmp/ica/bioinfo/Bi623/PS2/QAA/SRR25630305/SRR25630305_1.trim.unpaired.fastq.gz /projects/bgmp/ica/bioinfo/Bi623/PS2/QAA/SRR25630305/SRR25630305_2.trim.paired.fastq.gz /projects/bgmp/ica/bioinfo/Bi623/PS2/QAA/SRR25630305/SRR25630305_2.trim.unpaired.fastq.gz HEADCROP:8 LEADING:3 TRAILING:3 SLIDINGWINDOW:5:15 MINLEN:35
Quality encoding detected as phred33
Input Read Pairs: 4212076 Both Surviving: 4148753 (98.50%) Forward Only Surviving: 40471 (0.96%) Reverse Only Surviving: 19442 (0.46%) Dropped: 3410 (0.08%)
TrimmomaticPE: Completed successfully
	Command being timed: "trimmomatic PE -threads 4 /projects/bgmp/ica/bioinfo/Bi623/PS2/QAA/SRR25630305/SRR25630305_1.trim.fastq.gz /projects/bgmp/ica/bioinfo/Bi623/PS2/QAA/SRR25630305/SRR25630305_2.trim.fastq.gz /projects/bgmp/ica/bioinfo/Bi623/PS2/QAA/SRR25630305/SRR25630305_1.trim.paired.fastq.gz /projects/bgmp/ica/bioinfo/Bi623/PS2/QAA/SRR25630305/SRR25630305_1.trim.unpaired.fastq.gz /projects/bgmp/ica/bioinfo/Bi623/PS2/QAA/SRR25630305/SRR25630305_2.trim.paired.fastq.gz /projects/bgmp/ica/bioinfo/Bi623/PS2/QAA/SRR25630305/SRR25630305_2.trim.unpaired.fastq.gz HEADCROP:8 LEADING:3 TRAILING:3 SLIDINGWINDOW:5:15 MINLEN:35"
	User time (seconds): 256.55
	System time (seconds): 2.83
	Percent of CPU this job got: 345%
	Elapsed (wall clock) time (h:mm:ss or m:ss): 1:14.98
	Average shared text size (kbytes): 0
	Average unshared data size (kbytes): 0
	Average stack size (kbytes): 0
	Average total size (kbytes): 0
	Maximum resident set size (kbytes): 1164972
	Average resident set size (kbytes): 0
	Major (requiring I/O) page faults: 0
	Minor (reclaiming a frame) page faults: 315762
	Voluntary context switches: 22050
	Involuntary context switches: 5365
	Swaps: 0
	File system inputs: 0
	File system outputs: 128
	Socket messages sent: 0
	Socket messages received: 0
	Signals delivered: 0
	Page size (bytes): 4096
	Exit status: 0
	
	TrimmomaticPE: Started with arguments:
 -threads 4 /projects/bgmp/ica/bioinfo/Bi623/PS2/QAA/SRR25630397/SRR25630397_1.trim.fastq.gz /projects/bgmp/ica/bioinfo/Bi623/PS2/QAA/SRR25630397/SRR25630397_2.trim.fastq.gz /projects/bgmp/ica/bioinfo/Bi623/PS2/QAA/SRR25630397/SRR25630397_1.trim.paired.fastq.gz /projects/bgmp/ica/bioinfo/Bi623/PS2/QAA/SRR25630397/SRR25630397_1.trim.unpaired.fastq.gz /projects/bgmp/ica/bioinfo/Bi623/PS2/QAA/SRR25630397/SRR25630397_2.trim.paired.fastq.gz /projects/bgmp/ica/bioinfo/Bi623/PS2/QAA/SRR25630397/SRR25630397_2.trim.unpaired.fastq.gz HEADCROP:8 LEADING:3 TRAILING:3 SLIDINGWINDOW:5:15 MINLEN:35
Quality encoding detected as phred33
Input Read Pairs: 5623413 Both Surviving: 5555673 (98.80%) Forward Only Surviving: 46660 (0.83%) Reverse Only Surviving: 18937 (0.34%) Dropped: 2143 (0.04%)
TrimmomaticPE: Completed successfully
	Command being timed: "trimmomatic PE -threads 4 /projects/bgmp/ica/bioinfo/Bi623/PS2/QAA/SRR25630397/SRR25630397_1.trim.fastq.gz /projects/bgmp/ica/bioinfo/Bi623/PS2/QAA/SRR25630397/SRR25630397_2.trim.fastq.gz /projects/bgmp/ica/bioinfo/Bi623/PS2/QAA/SRR25630397/SRR25630397_1.trim.paired.fastq.gz /projects/bgmp/ica/bioinfo/Bi623/PS2/QAA/SRR25630397/SRR25630397_1.trim.unpaired.fastq.gz /projects/bgmp/ica/bioinfo/Bi623/PS2/QAA/SRR25630397/SRR25630397_2.trim.paired.fastq.gz /projects/bgmp/ica/bioinfo/Bi623/PS2/QAA/SRR25630397/SRR25630397_2.trim.unpaired.fastq.gz HEADCROP:8 LEADING:3 TRAILING:3 SLIDINGWINDOW:5:15 MINLEN:35"
	User time (seconds): 369.28
	System time (seconds): 6.32
	Percent of CPU this job got: 419%
	Elapsed (wall clock) time (h:mm:ss or m:ss): 1:29.46
	Average shared text size (kbytes): 0
	Average unshared data size (kbytes): 0
	Average stack size (kbytes): 0
	Average total size (kbytes): 0
	Maximum resident set size (kbytes): 1154084
	Average resident set size (kbytes): 0
	Major (requiring I/O) page faults: 0
	Minor (reclaiming a frame) page faults: 541804
	Voluntary context switches: 51414
	Involuntary context switches: 1143
	Swaps: 0
	File system inputs: 0
	File system outputs: 128
	Socket messages sent: 0
	Socket messages received: 0
	Signals delivered: 0
	Page size (bytes): 4096
	Exit status: 0

8. Plot the trimmed read length distributions for both paired R1 and paired R2 reads (on the same plot - yes, you will have to use Python or R to plot this. See ICA4 from Bi621). You can produce 2 different plots for your 2 different RNA-seq samples. There are a number of ways you could possibly do this. One useful thing your plot should show, for example, is whether R1s are trimmed more extensively than R2s, or vice versa. Comment on whether you expect R1s and R2s to be adapter-trimmed at different rates and why.

[Include your plot and comment on R1/R2 adapter trimming]

```{r, out.width='90%', fig.align='center'}

knitr::include_graphics("QAA_figs/Cco_com125_EO_6cm_trimmed_lengths.png")
knitr::include_graphics("QAA_figs/Crh_rhy107_EO_adult_trimmed_lengths.png")


```

R1 and R2 look basically the same. Both end up around 142 nt, so most trimming came from the fixed HEADCROP:8, not from cutting off adapters. In general, R1 and R2 should have about the same amount of adapter trimming because adapters show up when the DNA/RNA fragment is shorter than the read, which affects both reads equally. Any tiny differences are usually just because R2’s end is a bit lower quality, not because it has more adapters.

	Command being timed: "./plot_trimmed_lengths.py --sample Cco_com125_EO_6cm --r1 /projects/bgmp/ica/bioinfo/Bi623/PS2/QAA/SRR25630305/Cco_com125_EO_6cm_1.trim.paired.fastq.gz --r2 /projects/bgmp/ica/bioinfo/Bi623/PS2/QAA/SRR25630305/Cco_com125_EO_6cm_2.trim.paired.fastq.gz"
	User time (seconds): 13.83
	System time (seconds): 0.11
	Percent of CPU this job got: 98%
	Elapsed (wall clock) time (h:mm:ss or m:ss): 0:14.08
	Average shared text size (kbytes): 0
	Average unshared data size (kbytes): 0
	Average stack size (kbytes): 0
	Average total size (kbytes): 0
	Maximum resident set size (kbytes): 71840
	Average resident set size (kbytes): 0
	Major (requiring I/O) page faults: 0
	Minor (reclaiming a frame) page faults: 15148
	Voluntary context switches: 475
	Involuntary context switches: 65
	Swaps: 0
	File system inputs: 0
	File system outputs: 0
	Socket messages sent: 0
	Socket messages received: 0
	Signals delivered: 0
	Page size (bytes): 4096
	Exit status: 0
	


9. Bonus - Run `FastQC` on your trimmed data. Comment on differences you	Command being timed: "./plot_trimmed_lengths.py --sample Crh_rhy107_EO_adult --r1 /projects/bgmp/ica/bioinfo/Bi623/PS2/QAA/SRR25630397/Crh_rhy107_EO_adult_1.trim.paired.fastq.gz --r2 /projects/bgmp/ica/bioinfo/Bi623/PS2/QAA/SRR25630397/Crh_rhy107_EO_adult_2.trim.paired.fastq.gz"
	User time (seconds): 18.73
	System time (seconds): 0.20
	Percent of CPU this job got: 75%
	Elapsed (wall clock) time (h:mm:ss or m:ss): 0:24.96
	Average shared text size (kbytes): 0
	Average unshared data size (kbytes): 0
	Average stack size (kbytes): 0
	Average total size (kbytes): 0
	Maximum resident set size (kbytes): 72100
	Average resident set size (kbytes): 0
	Major (requiring I/O) page faults: 0
	Minor (reclaiming a frame) page faults: 15728
	Voluntary context switches: 2629
	Involuntary context switches: 106
	Swaps: 0
	File system inputs: 0
	File system outputs: 0
	Socket messages sent: 0
	Socket messages received: 0
	Signals delivered: 0
	Page size (bytes): 4096
	Exit status: 0 observe between the trimmed and untrimmed data. Include any figures needed to support your conclusions.

[Include command, comments on differences, and plot/s]

```{bash, eval=FALSE}



```

## Part 3 – Alignment and strand-specificity
10. Install additional software for alignment and counting of RNA-seq reads. In your QAA environment, use conda to install:
    - Star
    - Picard
    - Samtools
    - NumPy
    - Matplotlib
    - HTSeq

[Record details on how you installed these packages]

```{bash, eval=FALSE}

mamba install star

mamba install picard=2.18

mamba install samtools

mamba install numpy

mamba install matplotlib

mamba install htseq #numpy was downgraded to version 1.26.4

```

11. Download the publicly available *Campylomormyrus compressirostris* genome fasta and gff file from [Dryad](https://datadryad.org/dataset/doi:10.5061/dryad.c59zw3rcj) and generate an alignment database from it. If the download fails, the files are available `/projects/bgmp/shared/Bi623/PS2/campylomormyrus.fasta`, `/projects/bgmp/shared/Bi623/PS2/campylomormyrus.gff`. Align the reads to your *C. compressirostris* database using a splice-aware aligner. Use the settings specified in PS8 from Bi621. 

  > [!IMPORTANT]
  > You will need to use gene models to perform splice-aware alignment, see PS8 from Bi621. You may need to convert the gff file into a gtf file for this to work successfully.

[Record details on how you downloaded the genome, prepared the dataset for alignment, and commands for generating the alignment database and aligning reads]
    
```{bash, eval=FALSE}

##I was unable to download the files from Dryad due to an error (ERROR 403: Forbidden.)
cp /projects/bgmp/shared/Bi623/PS2/campylomormyrus.fasta .

cp /projects/bgmp/shared/Bi623/PS2/campylomormyrus.gff .

mamba install -c bioconda gffread -y

gffread -E campylomormyrus.gff -T -o- > campylomormyrus.gtf

STAR --runThreadN 8 \
    --runMode genomeGenerate \
    --genomeDir /projects/bgmp/ica/bioinfo/Bi623/PS2/QAA/SRR25630305/Campylomormyrus.Dryad.STAR_2.7.11b \
    --genomeFastaFiles ./campylomormyrus.fasta \
    --sjdbGTFfile ./campylomormyrus.gtf
    
STAR --runThreadN 8 --runMode alignReads \
    --outFilterMultimapNmax 3 \
    --outSAMunmapped Within KeepPairs \
    --alignIntronMax 1000000 --alignMatesGapMax 1000000 \
    --readFilesCommand zcat \
    --readFilesIn /projects/bgmp/ica/bioinfo/Bi623/PS2/QAA/SRR25630305/Cco_com125_EO_6cm_1.trim.paired.fastq.gz /projects/bgmp/ica/bioinfo/Bi623/PS2/QAA/SRR25630305/Cco_com125_EO_6cm_2.trim.paired.fastq.gz \
    --genomeDir /projects/bgmp/ica/bioinfo/Bi623/PS2/QAA/SRR25630305/Campylomormyrus.Dryad.STAR_2.7.11b \
    --outFileNamePrefix Cco_alignments
    
/usr/bin/time -v STAR --runThreadN 8 \
    --runMode genomeGenerate \
    --genomeDir /projects/bgmp/ica/bioinfo/Bi623/PS2/QAA/SRR25630397/Campylomormyrus.Dryad.STAR_2.7.11b \
    --genomeFastaFiles ./campylomormyrus.fasta \
    --sjdbGTFfile ./campylomormyrus.gtf
    
/usr/bin/time -v STAR --runThreadN 8 --runMode alignReads \
    --outFilterMultimapNmax 3 \
    --outSAMunmapped Within KeepPairs \
    --alignIntronMax 1000000 --alignMatesGapMax 1000000 \
    --readFilesCommand zcat \
    --readFilesIn /projects/bgmp/ica/bioinfo/Bi623/PS2/QAA/SRR25630397/Crh_rhy107_EO_adult_1.trim.paired.fastq.gz /projects/bgmp/ica/bioinfo/Bi623/PS2/QAA/SRR25630397/Crh_rhy107_EO_adult_2.trim.paired.fastq.gz \
    --genomeDir /projects/bgmp/ica/bioinfo/Bi623/PS2/QAA/SRR25630397/Campylomormyrus.Dryad.STAR_2.7.11b \
    --outFileNamePrefix Crh_alignments

```

!!!!! WARNING: --genomeSAindexNbases 14 is too large for the genome size=862592683, which may cause seg-fault at the mapping step. Re-run genome generation with recommended --genomeSAindexNbases 13
	Command being timed: "STAR --runThreadN 8 --runMode genomeGenerate --genomeDir /projects/bgmp/ica/bioinfo/Bi623/PS2/QAA/SRR25630305/Campylomormyrus.Dryad.STAR_2.7.11b --genomeFastaFiles ./campylomormyrus.fasta --sjdbGTFfile ./campylomormyrus.gtf"
	User time (seconds): 1361.42
	System time (seconds): 22.93
	Percent of CPU this job got: 276%
	Elapsed (wall clock) time (h:mm:ss or m:ss): 8:21.15
	Average shared text size (kbytes): 0
	Average unshared data size (kbytes): 0
	Average stack size (kbytes): 0
	Average total size (kbytes): 0
	Maximum resident set size (kbytes): 24998036
	Average resident set size (kbytes): 0
	Major (requiring I/O) page faults: 0
	Minor (reclaiming a frame) page faults: 14442625
	Voluntary context switches: 9654
	Involuntary context switches: 62395
	Swaps: 0
	File system inputs: 0
	File system outputs: 0
	Socket messages sent: 0
	Socket messages received: 0
	Signals delivered: 0
	Page size (bytes): 4096
	Exit status: 0
	
	Command being timed: "STAR --runThreadN 8 --runMode alignReads --outFilterMultimapNmax 3 --outSAMunmapped Within KeepPairs --alignIntronMax 1000000 --alignMatesGapMax 1000000 --readFilesCommand zcat --readFilesIn /projects/bgmp/ica/bioinfo/Bi623/PS2/QAA/SRR25630305/Cco_com125_EO_6cm_1.trim.paired.fastq.gz /projects/bgmp/ica/bioinfo/Bi623/PS2/QAA/SRR25630305/Cco_com125_EO_6cm_2.trim.paired.fastq.gz --genomeDir /projects/bgmp/ica/bioinfo/Bi623/PS2/QAA/SRR25630305/Campylomormyrus.Dryad.STAR_2.7.11b --outFileNamePrefix Cco_alignments"
	User time (seconds): 519.80
	System time (seconds): 4.00
	Percent of CPU this job got: 378%
	Elapsed (wall clock) time (h:mm:ss or m:ss): 2:18.21
	Average shared text size (kbytes): 0
	Average unshared data size (kbytes): 0
	Average stack size (kbytes): 0
	Average total size (kbytes): 0
	Maximum resident set size (kbytes): 11218316
	Average resident set size (kbytes): 0
	Major (requiring I/O) page faults: 0
	Minor (reclaiming a frame) page faults: 229342
	Voluntary context switches: 39388
	Involuntary context switches: 33847
	Swaps: 0
	File system inputs: 0
	File system outputs: 0
	Socket messages sent: 0
	Socket messages received: 0
	Signals delivered: 0
	Page size (bytes): 4096
	Exit status: 0
	
!!!!! WARNING: --genomeSAindexNbases 14 is too large for the genome size=862592683, which may cause seg-fault at the mapping step. Re-run genome generation with recommended --genomeSAindexNbases 13
	Command being timed: "STAR --runThreadN 8 --runMode genomeGenerate --genomeDir /projects/bgmp/ica/bioinfo/Bi623/PS2/QAA/SRR25630397/Campylomormyrus.Dryad.STAR_2.7.11b --genomeFastaFiles ./campylomormyrus.fasta --sjdbGTFfile ./campylomormyrus.gtf"
	User time (seconds): 1363.29
	System time (seconds): 24.38
	Percent of CPU this job got: 277%
	Elapsed (wall clock) time (h:mm:ss or m:ss): 8:20.25
	Average shared text size (kbytes): 0
	Average unshared data size (kbytes): 0
	Average stack size (kbytes): 0
	Average total size (kbytes): 0
	Maximum resident set size (kbytes): 23024808
	Average resident set size (kbytes): 0
	Major (requiring I/O) page faults: 0
	Minor (reclaiming a frame) page faults: 15266420
	Voluntary context switches: 9913
	Involuntary context switches: 62511
	Swaps: 0
	File system inputs: 0
	File system outputs: 0
	Socket messages sent: 0
	Socket messages received: 0
	Signals delivered: 0
	Page size (bytes): 4096
	Exit status: 0
	
	Command being timed: "STAR --runThreadN 8 --runMode alignReads --outFilterMultimapNmax 3 --outSAMunmapped Within KeepPairs --alignIntronMax 1000000 --alignMatesGapMax 1000000 --readFilesCommand zcat --readFilesIn /projects/bgmp/ica/bioinfo/Bi623/PS2/QAA/SRR25630397/Crh_rhy107_EO_adult_1.trim.paired.fastq.gz /projects/bgmp/ica/bioinfo/Bi623/PS2/QAA/SRR25630397/Crh_rhy107_EO_adult_2.trim.paired.fastq.gz --genomeDir /projects/bgmp/ica/bioinfo/Bi623/PS2/QAA/SRR25630397/Campylomormyrus.Dryad.STAR_2.7.11b --outFileNamePrefix Crh_alignments"
	User time (seconds): 456.07
	System time (seconds): 4.24
	Percent of CPU this job got: 373%
	Elapsed (wall clock) time (h:mm:ss or m:ss): 2:03.27
	Average shared text size (kbytes): 0
	Average unshared data size (kbytes): 0
	Average stack size (kbytes): 0
	Average total size (kbytes): 0
	Maximum resident set size (kbytes): 11229164
	Average resident set size (kbytes): 0
	Major (requiring I/O) page faults: 0
	Minor (reclaiming a frame) page faults: 219898
	Voluntary context switches: 54338
	Involuntary context switches: 35251
	Swaps: 0
	File system inputs: 0
	File system outputs: 0
	Socket messages sent: 0
	Socket messages received: 0
	Signals delivered: 0
	Page size (bytes): 4096
	Exit status: 0

12. Remove PCR duplicates using [Picard MarkDuplicates](https://broadinstitute.github.io/picard/command-line-overview.html#MarkDuplicates). You may need to sort your reads with `samtools` before running Picard. 
- Use the following for running picard: picard MarkDuplicates INPUT=[FILE] OUTPUT=[FILE] METRICS_FILE=[FILENAME].metrics REMOVE_DUPLICATES=TRUE VALIDATION_STRINGENCY=LENIENT
 

```{bash, eval=FALSE}

samtools sort -O SAM -o Cco_alignmentsAligned.out.sorted.sam Cco_alignmentsAligned.out.sam

samtools sort -O SAM -o Crh_alignmentsAligned.out.sorted.sam Crh_alignmentsAligned.out.sam

picard MarkDuplicates \
  INPUT=/projects/bgmp/ica/bioinfo/Bi623/PS2/QAA/SRR25630305/Cco_com125_EO_6cm.out.sorted.sam \
  OUTPUT=/projects/bgmp/ica/bioinfo/Bi623/PS2/QAA/SRR25630305/Cco_com125_EO_6cm.out.sorted.rmdup.sam \
  METRICS_FILE=/projects/bgmp/ica/bioinfo/Bi623/PS2/QAA/SRR25630305/Cco_com125_EO_6cm.metrics \
  REMOVE_DUPLICATES=TRUE \
  VALIDATION_STRINGENCY=LENIENT
  
picard MarkDuplicates \
  INPUT=/projects/bgmp/ica/bioinfo/Bi623/PS2/QAA/SRR25630397/Crh_rhy107_EO_adult.out.sorted.sam \
  OUTPUT=/projects/bgmp/ica/bioinfo/Bi623/PS2/QAA/SRR25630397/Crh_rhy107_EO_adult.out.sorted.rmdup.sam \
  METRICS_FILE=/projects/bgmp/ica/bioinfo/Bi623/PS2/QAA/SRR25630397/Crh_rhy107_EO_adult.metrics \
  REMOVE_DUPLICATES=TRUE \
  VALIDATION_STRINGENCY=LENIENT

```

	Command being timed: "picard MarkDuplicates INPUT=/projects/bgmp/ica/bioinfo/Bi623/PS2/QAA/SRR25630305/Cco_com125_EO_6cm.out.sorted.sam OUTPUT=/projects/bgmp/ica/bioinfo/Bi623/PS2/QAA/SRR25630305/Cco_com125_EO_6cm.out.sorted.rmdup.sam METRICS_FILE=/projects/bgmp/ica/bioinfo/Bi623/PS2/QAA/SRR25630305/Cco_com125_EO_6cm.metrics REMOVE_DUPLICATES=TRUE VALIDATION_STRINGENCY=LENIENT"
	User time (seconds): 84.96
	System time (seconds): 14.82
	Percent of CPU this job got: 116%
	Elapsed (wall clock) time (h:mm:ss or m:ss): 1:25.39
	Average shared text size (kbytes): 0
	Average unshared data size (kbytes): 0
	Average stack size (kbytes): 0
	Average total size (kbytes): 0
	Maximum resident set size (kbytes): 965716
	Average resident set size (kbytes): 0
	Major (requiring I/O) page faults: 0
	Minor (reclaiming a frame) page faults: 126748
	Voluntary context switches: 20190
	Involuntary context switches: 468
	Swaps: 0
	File system inputs: 0
	File system outputs: 130424
	Socket messages sent: 0
	Socket messages received: 0
	Signals delivered: 0
	Page size (bytes): 4096
	Exit status: 0
	
	Command being timed: "picard MarkDuplicates INPUT=/projects/bgmp/ica/bioinfo/Bi623/PS2/QAA/SRR25630397/Crh_rhy107_EO_adult.out.sorted.sam OUTPUT=/projects/bgmp/ica/bioinfo/Bi623/PS2/QAA/SRR25630397/Crh_rhy107_EO_adult.out.sorted.rmdup.sam METRICS_FILE=/projects/bgmp/ica/bioinfo/Bi623/PS2/QAA/SRR25630397/Crh_rhy107_EO_adult.metrics REMOVE_DUPLICATES=TRUE VALIDATION_STRINGENCY=LENIENT"
	User time (seconds): 111.99
	System time (seconds): 19.38
	Percent of CPU this job got: 117%
	Elapsed (wall clock) time (h:mm:ss or m:ss): 1:51.36
	Average shared text size (kbytes): 0
	Average unshared data size (kbytes): 0
	Average stack size (kbytes): 0
	Average total size (kbytes): 0
	Maximum resident set size (kbytes): 1101676
	Average resident set size (kbytes): 0
	Major (requiring I/O) page faults: 0
	Minor (reclaiming a frame) page faults: 155259
	Voluntary context switches: 36335
	Involuntary context switches: 657
	Swaps: 0
	File system inputs: 0
	File system outputs: 358720
	Socket messages sent: 0
	Socket messages received: 0
	Signals delivered: 0
	Page size (bytes): 4096
	Exit status: 0

13. Using your script from PS8 in Bi621, report the number of mapped and unmapped reads from each of your 2 SAM files post deduplication with picard. Make sure that your script is looking at the bitwise flag to determine if reads are primary or secondary mapping (update/fix your script if necessary).

For Cco_com125_EO_6cm.out.sorted.rmdup.sam:
Number of unique mapped reads: 1903088
Number of unique unmapped reads: 958755

For Crh_rhy107_EO_adult.out.sorted.rmdup.sam:
Number of unique mapped reads: 2855485
Number of unique unmapped reads: 913124

14. Count deduplicated reads that map to features using `htseq-count`. You should run htseq-count twice: once with `--stranded=yes` and again with `--stranded=reverse`. Use default parameters otherwise. You may need to use the `-i` parameter for this run.

```{bash, eval=FALSE}

/usr/bin/time -v htseq-count -s yes -i gene_id\
  /projects/bgmp/ica/bioinfo/Bi623/PS2/QAA/SRR25630305/Cco_com125_EO_6cm.out.sorted.rmdup.sam \
  /projects/bgmp/ica/bioinfo/Bi623/PS2/QAA/SRR25630305/campylomormyrus.gtf \
  > /projects/bgmp/ica/bioinfo/Bi623/PS2/QAA/SRR25630305/Cco_com125_EO_6cm.htseq.stranded_yes.txt

/usr/bin/time -v htseq-count -s reverse -i gene_id\
  /projects/bgmp/ica/bioinfo/Bi623/PS2/QAA/SRR25630305/Cco_com125_EO_6cm.out.sorted.rmdup.sam \
  /projects/bgmp/ica/bioinfo/Bi623/PS2/QAA/SRR25630305/campylomormyrus.gtf \
  > /projects/bgmp/ica/bioinfo/Bi623/PS2/QAA/SRR25630305/Cco_com125_EO_6cm.htseq.stranded_reverse.txt
  

```

	Command being timed: "htseq-count -s yes -i gene_id /projects/bgmp/ica/bioinfo/Bi623/PS2/QAA/SRR25630305/Cco_com125_EO_6cm.out.sorted.rmdup.sam /projects/bgmp/ica/bioinfo/Bi623/PS2/QAA/SRR25630305/campylomormyrus.gtf"
	User time (seconds): 135.61
	System time (seconds): 0.95
	Percent of CPU this job got: 99%
	Elapsed (wall clock) time (h:mm:ss or m:ss): 2:17.12
	Average shared text size (kbytes): 0
	Average unshared data size (kbytes): 0
	Average stack size (kbytes): 0
	Average total size (kbytes): 0
	Maximum resident set size (kbytes): 153724
	Average resident set size (kbytes): 0
	Major (requiring I/O) page faults: 0
	Minor (reclaiming a frame) page faults: 140402
	Voluntary context switches: 1494
	Involuntary context switches: 59
	Swaps: 0
	File system inputs: 0
	File system outputs: 0
	Socket messages sent: 0
	Socket messages received: 0
	Signals delivered: 0
	Page size (bytes): 4096
	Exit status: 0
	
Command being timed: "htseq-count -s reverse -i gene_id /projects/bgmp/ica/bioinfo/Bi623/PS2/QAA/SRR25630305/Cco_com125_EO_6cm.out.sorted.rmdup.sam /projects/bgmp/ica/bioinfo/Bi623/PS2/QAA/SRR25630305/campylomormyrus.gtf"
	User time (seconds): 146.92
	System time (seconds): 0.93
	Percent of CPU this job got: 99%
	Elapsed (wall clock) time (h:mm:ss or m:ss): 2:28.13
	Average shared text size (kbytes): 0
	Average unshared data size (kbytes): 0
	Average stack size (kbytes): 0
	Average total size (kbytes): 0
	Maximum resident set size (kbytes): 151708
	Average resident set size (kbytes): 0
	Major (requiring I/O) page faults: 0
	Minor (reclaiming a frame) page faults: 130322
	Voluntary context switches: 390
	Involuntary context switches: 59
	Swaps: 0
	File system inputs: 0
	File system outputs: 0
	Socket messages sent: 0
	Socket messages received: 0
	Signals delivered: 0
	Page size (bytes): 4096
	Exit status: 0
	
	Command being timed: "htseq-count -s yes -i gene_id /projects/bgmp/ica/bioinfo/Bi623/PS2/QAA/SRR25630397/Crh_rhy107_EO_adult.out.sorted.rmdup.sam /projects/bgmp/ica/bioinfo/Bi623/PS2/QAA/SRR25630397/campylomormyrus.gtf"
	User time (seconds): 190.78
	System time (seconds): 1.11
	Percent of CPU this job got: 99%
	Elapsed (wall clock) time (h:mm:ss or m:ss): 3:12.40
	Average shared text size (kbytes): 0
	Average unshared data size (kbytes): 0
	Average stack size (kbytes): 0
	Average total size (kbytes): 0
	Maximum resident set size (kbytes): 151776
	Average resident set size (kbytes): 0
	Major (requiring I/O) page faults: 1
	Minor (reclaiming a frame) page faults: 140667
	Voluntary context switches: 328
	Involuntary context switches: 81
	Swaps: 0
	File system inputs: 0
	File system outputs: 0
	Socket messages sent: 0
	Socket messages received: 0
	Signals delivered: 0
	Page size (bytes): 4096
	Exit status: 0
	
	Command being timed: "htseq-count -s reverse -i gene_id /projects/bgmp/ica/bioinfo/Bi623/PS2/QAA/SRR25630397/Crh_rhy107_EO_adult.out.sorted.rmdup.sam /projects/bgmp/ica/bioinfo/Bi623/PS2/QAA/SRR25630397/campylomormyrus.gtf"
	User time (seconds): 191.95
	System time (seconds): 1.03
	Percent of CPU this job got: 99%
	Elapsed (wall clock) time (h:mm:ss or m:ss): 3:13.45
	Average shared text size (kbytes): 0
	Average unshared data size (kbytes): 0
	Average stack size (kbytes): 0
	Average total size (kbytes): 0
	Maximum resident set size (kbytes): 151780
	Average resident set size (kbytes): 0
	Major (requiring I/O) page faults: 0
	Minor (reclaiming a frame) page faults: 145829
	Voluntary context switches: 452
	Involuntary context switches: 61
	Swaps: 0
	File system inputs: 0
	File system outputs: 0
	Socket messages sent: 0
	Socket messages received: 0
	Signals delivered: 0
	Page size (bytes): 4096
	Exit status: 0

15. Demonstrate convincingly whether or not the data are from "strand-specific" RNA-Seq libraries **and** which `stranded=` parameter should you use for counting your reads for a future differential gene expression analyses. Include any commands/scripts used. Briefly describe your evidence, using quantitative statements (e.g. "I propose that these data are/are not strand-specific, because X% of the reads are y, as opposed to z."). This [kit](https://www.revvity.com/product/nex-rapid-dir-rna-seq-kit-2-0-8rxn-nova-5198-01) was used during library preparation. This [paper](https://academic.oup.com/bfg/article/19/5-6/339/5837822) may provide helpful information.

  > [!TIP]
  > Recall ICA4 from Bi621.

[Describe whether your reads are "string-specific", why you think they are, any evidence, and which stranded parameter is appropriate and why]

| calc |  assigned |     total | percent |
| ---: | --------: | --------: | :-----: |
|    1 |    93,309 | 4,508,188 |  2.07%  |
|    2 | 1,942,036 | 4,508,188 |  43.08% |
|    3 |   153,957 | 6,360,672 |  2.42%  |
|    4 | 3,344,574 | 6,360,672 |  52.58% |


These data are strand-specific (reverse-stranded). With htseq-count, --stranded=reverse assigned 43.08% of reads to genes vs 2.07% with --stranded=yes for the Cco_com125_EO_6cm dataset. With htseq-count, --stranded=reverse assigned 52.58% of reads to genes vs 2.42% with --stranded=yes for the Crh_rhy107_EO_adult dataset. Given the kit’s dUTP chemistry, this matches expectations. We will use htseq-count --stranded=reverse for downstream differential expression. 

16. BONUS - Turn your commands from part 1 and 2 into a script with a loop going through your two SRA files

## Bonus (optional!)

Review the [publication](https://doi.org/10.1093/molbev/msae021) from PRJNA1005244 or the third chapter of the [thesis](https://canvas.uoregon.edu/courses/266187/files/22059308?module_item_id=5380118) for the PRJNA1005245 dataset. See if this information leads to any additional insight of your analysis.

[Add insights to the dataset]

## Upload your:
- [ ] lab notebook
- [ ] Talapas batch script/code
- [ ] FastQC plots
- [ ] counts files generated from htseq-count (in a folder would be nice; **only include the counts files that would be used in a future differential RNA-seq analysis: use the format Species_sample_tissue_age/size_sample#_readnumber_htseqcounts_[revORyes]stranded.txt**)
- [ ] pdf report (see below; turn into both Github AND Canvas)
- [ ] and any additional plots, code, or code output

to GitHub.
    
### Pdf report details
You should create a pdf file (using Rmarkdown) with a high-level report including:

- [ ] all requested plots
- [ ] answers to questions
- [ ] mapped/unmapped read counts from PS8 script (in a nicely formatted table)
- [ ] It should be named `QAA_report.pdf`
- [ ] Include at the top level of your repo
- [ ] ALSO, submit it to Canvas.

> [!TIP]
> You may need to install LaTeX to knit your rmarkdown into a pdf file. Run `tinytex::install_tinytex()` to install it on R.
   
The three parts of the assignment should be clearly labeled. Be sure to title and write a descriptive figure caption for each image/graph/table you present. 

> [!TIP]
> Think about figure captions you've read and discussed in Journal Club. Find some good examples to model your captions on.
